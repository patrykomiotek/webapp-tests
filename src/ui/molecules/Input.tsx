import { ComponentPropsWithoutRef, useId } from 'react';

type Props = ComponentPropsWithoutRef<'input'> & {
  label: string;
};

export const Input = ({ label, className, ...rest }: Props) => {
  const id = useId();
  return (
    <div>
      <label htmlFor={id}>{label}</label>
      <input id={id} className={`my-btn ${className}`} {...rest} />
    </div>
  );
};
