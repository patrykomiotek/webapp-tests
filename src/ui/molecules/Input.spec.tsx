import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import { Input } from './Input';

describe('input component', () => {
  it('should match snapshot', () => {
    const { container } = render(<Input label="My input" />);
    // console.log({ container });
    expect(container).toMatchSnapshot();
  });

  it('should pass accessibility check', async () => {
    const { container } = render(<Input label="My input" />);

    const result = await axe(container);

    expect(result).toHaveNoViolations();
  });
});
