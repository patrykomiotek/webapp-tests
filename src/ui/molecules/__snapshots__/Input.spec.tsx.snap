// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`input component > should match snapshot 1`] = `
<div>
  <div>
    <label
      for=":r0:"
    >
      My input
    </label>
    <input
      class="my-btn undefined"
      id=":r0:"
    />
  </div>
</div>
`;
